scripts/kafka-integration-test.sh

```diff
@@ -3,48 +3,37 @@
 set -e
 
 export STORAGE=kafka
-
-# Function to start Kafka
-start_kafka() {
-    echo "Starting Kafka..."
-    
-    docker run --name kafka -d \
-    -p 9092:9092 \
-    -e KAFKA_CFG_NODE_ID=0 \
-    -e KAFKA_CFG_PROCESS_ROLES=controller,broker \
-    -e KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@localhost:9093 \
-    -e KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093 \
-    -e KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://localhost:9092 \
-    -e KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT \
-    -e KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER \
-    -e KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT \
-    bitnami/kafka:3.6.0
-}
-
+compose_file="docker-compose/kafka-integration-test/v3.yml"
 # Check if the -k parameter is provided or not
 if [ "$1" == "-k" ]; then
-    start_kafka
+    echo "Starting Kafka using Docker Compose..."
+    docker compose -f $compose_file up -d kafka
 fi
 
 # Set the timeout in seconds
 timeout=180
 # Set the interval between checks in seconds
 interval=5
 
+# Function to check if Kafka is ready
+is_kafka_ready() {
+    docker compose -f $compose_file exec kafka /opt/bitnami/kafka/bin/kafka-topics.sh --list --bootstrap-server localhost:9092 >/dev/null 2>&1
```

Comment:
nit : add line breaks for better readability
```suggestion
    docker compose -f $compose_file \
        exec kafka /opt/bitnami/kafka/bin/kafka-topics.sh \
        --list --bootstrap-server localhost:9092 \
        >/dev/null 2>&1
```